src/webapp/reports/autogenerated-forms/GridFormViewModel.ts

```diff
@@ -56,10 +59,29 @@ export class GridViewModel {
             .map(
                 ([groupName, dataElementsForGroup]): SubSectionGrid => ({
                     name: groupName,
-                    dataElements: dataElementsForGroup.map(dataElement => ({
-                        ...dataElement,
-                        name: _(dataElement.name).split(separator).last() || "-",
-                    })),
+                    dataElements: dataElementsForGroup.flatMap(dataElement => {
```

Comment:
Not sure why we're changing the way subsections are being calculated? Looks like is not related to the calculateTotals process.

Reply:
you're right. this was done while i was trying to figure out the task. i will revert the change. thank you!